ons_alpha/topics/models.py

```diff
@@ -1,15 +1,81 @@
+from django.core.exceptions import ValidationError
+from django.db import models
+from wagtail.admin.panels import FieldPanel, ObjectList, TabbedInterface
+
 from ons_alpha.core.models.base import BasePage
 from ons_alpha.core.models.mixins import SubpageMixin
+from ons_alpha.taxonomy.models import Topic
+from ons_alpha.taxonomy.views import topic_page_topic_chooser_viewset, topic_section_page_topic_chooser_viewset
 
 
 class TopicPage(SubpageMixin, BasePage):
+    _taxonomy_widget_class = topic_page_topic_chooser_viewset.widget_class
+
     template = "templates/pages/topics/topic_page.html"
     parent_page_types = ["topics.TopicSectionPage"]
     subpage_types = ["bulletins.BulletinSeriesPage"]
-    page_description = "A specific topic page. e.g.  Public sector finance or  Inflation and price indices"
+    page_description = "A specific topic page. e.g. Public sector finance or Inflation and price indices"
+
+    topic = models.OneToOneField(
+        Topic,
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=False,
+        related_name="topic_page",
+    )
+
+    content_panels = BasePage.content_panels
+
+    edit_handler = TabbedInterface(
```

Comment:
nice use of the tabbed inderface.

I'd suggest setting the widget class directly in the field panel definition, rather than via `_taxonomy_widget_class` thought. We don't need it outside of that. Also, if we keep to the one chooser, then you don't need to specify the widget at all

Reply:
Only a single chooser now so removed the need to pass widget class.